phenol-annotations/src/main/java/org/monarchinitiative/phenol/annotations/formats/hpo/category/HpoCategoryLookup.java

```diff
@@ -0,0 +1,87 @@
+package org.monarchinitiative.phenol.annotations.formats.hpo.category;
+
+import org.monarchinitiative.phenol.graph.OntologyGraph;
+import org.monarchinitiative.phenol.ontology.data.Term;
+import org.monarchinitiative.phenol.ontology.data.TermId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class HpoCategoryLookup {
```

Comment:
Two points wrt. this class.

I think it would be better to make this class to just work with the `TermId`s. Then less information is hard-coded (no labels). This assumes that the user of this class has `MinimalOntology` around, and can translate the `TermId` into its label (EN, CZ, NE, whatever language). In result `getPrioritizedCategory` method will return `List<TermId>` instead of `Term`s.

Second, I think it would be better to ask for the categories via the constructor and not hard-code them as a field. This will help us change the categories in the future, and make the class more usable, in general. Asking for a `List<TermId>` is probably OK, I am not sure having `TermId[]` has any advantages. In result, the `HpoCategoryLookup` will have two private final fields: `OntologyGraph<TermId> graph` and `List<TermId> categories`.